### PR TITLE
interfaces/hotplug: renamed RequestedSlotSpec to ProposedSlot, removed Specification

### PR DESCRIPTION
**Pull request by stolowski**
As agreed, removed hotplug.Specification and simplified it to just ProposedSlot that replaces RequestedSlotSpec. The new Clean() method will be called by hotplug machinery to clean/validate proposed slots returned by interfaces (not in this PR).

### CODE REVIEW

File: interfaces/hotplug/proposed_slot.go [hunk 1]
```diff
@@ -47,43 +44,25 @@ type RequestedSlotSpec struct {
...
 	if attrs == nil {
 		attrs = make(map[string]interface{})
 	} else {
-		attrs = utils.CopyAttributes(slotSpec.Attrs)
-	}
-	h.slot = &RequestedSlotSpec{
-		Name:  slotSpec.Name,
-		Label: slotSpec.Label,
-		Attrs: utils.NormalizeInterfaceAttributes(attrs).(map[string]interface{}),
+		attrs = utils.CopyAttributes(slot.Attrs)
```
Comment (pedronis) [inline interfaces/hotplug/proposed_slot.go]: doesn't NormalizeInterfaceAttributes essentially do a copy too? do we need this copy?

Comment (stolowski) [reply interfaces/hotplug/proposed_slot.go]: Ah, of course, you're right, fixed. Thanks!

File: overlord/ifacestate/ifacestate_test.go [hunk 2]
```diff
@@ -5644,8 +5644,8 @@ func (s *interfaceManagerSuite) TestHotplugAddGoneSlot(c *C) {
 	t := s.state.NewTask("hotplug-add-slot", "")
 	t.Set("hotplug-key", "1234")
 	t.Set("interface", "test")
-	spec := hotplug.RequestedSlotSpec{Name: "hotplugslot", Label: "", Attrs: map[string]interface{}{"foo": "bar"}}
-	t.Set("slot-spec", spec)
+	spec := hotplug.ProposedSlot{Name: "hotplugslot", Label: "", Attrs: map[string]interface{}{"foo": "bar"}}
```
Comment (mvo5) [inline overlord/ifacestate/ifacestate_test.go]: Quick drive-by: maybe use a different name here than `spec` ?

Comment (stolowski) [reply overlord/ifacestate/ifacestate_test.go]: Done.

File: interfaces/hotplug/proposed_slot_test.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
```
Comment (mvo5) [inline interfaces/hotplug/proposed_slot_test.go]: This should be 2019 now I think.

File: interfaces/hotplug/proposed_slot_test.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+package hotplug
+
+import (
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type proposedSlotSuite struct {
```
Comment (mvo5) [inline interfaces/hotplug/proposed_slot_test.go]: Nice tests!
